src/Aspire.Hosting.Azure/Provisioning/Internal/RunModeProvisioningContextProvider.cs

```diff
@@ -0,0 +1,145 @@
+#pragma warning disable ASPIREINTERACTION001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Text.Json.Nodes;
+using Aspire.Hosting.Azure.Resources;
+using Aspire.Hosting.Azure.Utils;
+using Aspire.Hosting.Publishing;
+using Azure.Core;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Hosting.Azure.Provisioning.Internal;
+
+/// <summary>
+/// Run mode implementation of <see cref="IProvisioningContextProvider"/>.
+/// </summary>
+internal sealed class RunModeProvisioningContextProvider(
+    IInteractionService interactionService,
+    IOptions<AzureProvisionerOptions> options,
+    IHostEnvironment environment,
+    ILogger<RunModeProvisioningContextProvider> logger,
+    IArmClientProvider armClientProvider,
+    IUserPrincipalProvider userPrincipalProvider,
+    ITokenCredentialProvider tokenCredentialProvider,
+    DistributedApplicationExecutionContext distributedApplicationExecutionContext,
+    IOptions<PublishingOptions> publishingOptions) : BaseProvisioningContextProvider(
+        interactionService,
+        options,
+        environment,
+        logger,
+        armClientProvider,
+        userPrincipalProvider,
+        tokenCredentialProvider,
+        distributedApplicationExecutionContext,
+        publishingOptions)
+{
+    protected override string GetDefaultResourceGroupName()
+    {
+        var prefix = "rg-aspire";
+
+        if (!string.IsNullOrWhiteSpace(_options.ResourceGroupPrefix))
+        {
+            prefix = _options.ResourceGroupPrefix;
+        }
+
+        var suffix = RandomNumberGenerator.GetHexString(8, lowercase: true);
+
+        var maxApplicationNameSize = ResourceGroupNameHelpers.MaxResourceGroupNameLength - prefix.Length - suffix.Length - 2; // extra '-'s
+
+        var normalizedApplicationName = ResourceGroupNameHelpers.NormalizeResourceGroupName(_environment.ApplicationName.ToLowerInvariant());
+        if (normalizedApplicationName.Length > maxApplicationNameSize)
+        {
+            normalizedApplicationName = normalizedApplicationName[..maxApplicationNameSize];
+        }
+
+        // Run mode always includes random suffix for uniqueness
+        return $"{prefix}-{normalizedApplicationName}-{suffix}";
+    }
+
+    protected override async Task RetrieveAzureProvisioningOptions(JsonObject userSecrets, CancellationToken cancellationToken = default)
+    {
+        var locations = typeof(AzureLocation).GetProperties(BindingFlags.Public | BindingFlags.Static)
```

Comment:
We are still going to want to change this logic to match what deployment does. (prompt for sub, then locations)

Reply:
In separate prompts?  I don't want to break up the prompting experience in the dashboard until we've figured out async resolution of choices.